ssl/record/ssl3_record_tls13.c

```diff
@@ -52,8 +53,18 @@ int tls13_enc(SSL *s, SSL3_RECORD *recs, size_t n_recs, int send)
         rec->input = rec->data;
         return 1;
     }
+
     ivlen = EVP_CIPHER_CTX_iv_length(ctx);
 
+    /*
+     * To get here we must have selected a ciphersuite - otherwise ctx would
+     * be NULL
+     */
+    assert(s->s3->tmp.new_cipher != NULL);
+    if (s->s3->tmp.new_cipher == NULL)
+        return -1;
+    alg_enc = s->s3->tmp.new_cipher->algorithm_enc;
```

Comment:
The function tls13_enc() gets called even if we are not going to encrypt anything (which means we may not have selected a ciphersuite yet), i.e. if we're still sending records in plaintext. In that case we just do a memove of the input data to the output. This happens in the `if(ctx == NULL)` block above. In that case the ciphersuite may not have been set up yet. So we need to make sure we only deref `new_cipher` after the `ctx == NULL` check.